packages/core/admin/admin/src/content-manager/components/AttributeFilter/index.js

```diff
@@ -1,13 +1,87 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
+import { useQueryParams, getDisplayName } from '@strapi/helper-plugin';
+import { useQueryClient } from 'react-query';
+
+import { useAdminUsers } from '../../../hooks/useAdminUsers';
 import useAllowedAttributes from './hooks/useAllowedAttributes';
 import Filters from './Filters';
+import AdminUsersFilter from './AdminUsersFilter';
+
+const customOperators = [
+  {
+    intlLabel: { id: 'components.FilterOptions.FILTER_TYPES.$eq', defaultMessage: 'is' },
+    value: '$eq',
+  },
+  {
+    intlLabel: { id: 'components.FilterOptions.FILTER_TYPES.$ne', defaultMessage: 'is not' },
+    value: '$ne',
+  },
+];
+
+const AUTHOR_ATTRIBUTES = ['createdBy', 'updatedBy'];
+
+const getUsersSelected = (query) => {
+  return (
+    query?.filters?.$and?.reduce((acc, filter) => {
+      const [key, value] = Object.entries(filter)[0];
+      const id = value.id?.$eq || value.id?.$ne;
+
+      if (AUTHOR_ATTRIBUTES.includes(key) && !acc.includes(id)) {
+        acc.push(id);
+      }
+
+      return acc;
+    }, []) ?? []
+  );
+};
+
+const formatUsers = (users, formatMessage) => {
+  return users.map((user) => ({
+    label: getDisplayName(user, formatMessage),
+    customValue: user.id.toString(),
+  }));
+};
 
 const AttributeFilter = ({ contentType, slug, metadatas }) => {
   const { formatMessage } = useIntl();
   const allowedAttributes = useAllowedAttributes(contentType, slug);
+  const queryClient = useQueryClient();
+  const [{ query }] = useQueryParams();
+  const selectedUsers = getUsersSelected(query);
+  const { users, isLoading } = useAdminUsers(
+    { filter: { id: { in: selectedUsers } } },
+    {
+      enabled:
+        queryClient.getQueryData(['users', '', {}]) === undefined && selectedUsers.length > 0,
+    }
+  );
+
   const displayedFilters = allowedAttributes.map((name) => {
+    if (AUTHOR_ATTRIBUTES.includes(name)) {
+      return {
+        name,
+        metadatas: {
+          label: formatMessage({
+            id: `content-manager.components.Filters.${name}`,
+            defaultMessage: name,
+          }),
+          customOperators,
+          customInput: AdminUsersFilter,
+          options: formatUsers(
+            queryClient.getQueryData(['users', '', {}])?.results ?? users,
```

Comment:
I worry this is very brittle and relies on us having called the base `useAdminUsers` with the params as we expect imho we should just have two calls to `useAdminUsers`...

Reply:
well in theory it will just use the cache if the hook was called higher in the tree – you're basically bypassing the hook to access the cache directly whilst potentially opening yourself to their being no cache imo.

Did you read this was the better way to do it somewhere? :)